### PR TITLE
Enhance reusable SiteOrigin component

### PR DESCRIPTION
**Pull request by adonesky1**
Make existing component `<SiteOrigin>` reusable across all contexts where we surface a connected site's origin. Now boolean typed `chip` prop can be added or omitted so that this component can be a chip or just a plain span with standardized formatting for displaying an origin.

To test click through all actions on the test-dapp and look for the site origin and make sure that it is correctly formatted.

### CODE REVIEW

File: ui/components/ui/site-origin/site-origin.js [hunk 1]
```diff
@@ -1,16 +1,35 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import classnames from 'classnames';
 import Chip from '../chip';
 import IconWithFallback from '../icon-with-fallback';
 
-export default function SiteOrigin({ siteOrigin, iconSrc, iconName }) {
+export default function SiteOrigin({
+  siteOrigin,
```
Comment (darkwing) [inline ui/components/ui/site-origin/site-origin.js:8]: `siteOrigin` is used in both rendering cases, so I think we should make it `required` in `props` validation.

File: ui/components/app/signature-request/index.scss [hunk 2]
```diff
@@ -68,6 +68,9 @@
 
   &__info {
     @include H7;
+
+    max-width: 90%;
```
Comment (darkwing) [inline ui/components/app/signature-request/index.scss:72]: Where does this number come from?

**Review by darkwing — DISMISSED** [review]
Just a few items.  I love that we're using this everywhere though!

File: ui/components/ui/site-origin/site-origin.js (hunk 1 above)
Comment (adonesky1) [reply ui/components/ui/site-origin/site-origin.js:8]: 🤔  I think it is already required? [line 38](https://github.com/MetaMask/metamask-extension/pull/14720/files#diff-518115d24597fde0bd8b5e57f626086ac81d13595fc3b90335e2972ae76e654cR38)

File: app/_locales/en/messages.json [hunk 3]
```diff
@@ -1057,8 +1057,7 @@
     "description": "$1 is a token symbol, e.g. ETH"
   },
   "encryptionPublicKeyNotice": {
-    "message": "$1 would like your public encryption key. By consenting, this site will be able to compose encrypted messages to you.",
-    "description": "$1 is the web3 site name"
+    "message": "would like your public encryption key. By consenting, this site will be able to compose encrypted messages to you."
```
[the remark this answers is not in the document]
Comment (adonesky1) [reply app/_locales/en/messages.json:1060]: Yes totally. I thought I needed to this for a formatting reason but it turns out I don't even need to. Plus you're right I hadn't thought about locale issues.

File: ui/components/app/signature-request/index.scss (hunk 2 above)
Comment (adonesky1) [reply ui/components/app/signature-request/index.scss:72]: currently on signedTypeData requests from sufficiently long origins the ends of the address butt up against the sides of the notification. But I think you're right we shouldn't use an arbitrary width value here. I think it makes more sense to add left and right padding! So I'll switch to that.

File: ui/components/ui/site-origin/site-origin.js (hunk 1 above)
Comment (darkwing) [reply ui/components/ui/site-origin/site-origin.js:8]: d'oh, sorry for not seeing that!

**Review by digiwand — DISMISSED** [review]
LGTM!

File: ui/components/ui/site-origin/site-origin.js [hunk 4]
```diff
@@ -19,4 +38,7 @@ SiteOrigin.propTypes = {
   siteOrigin: PropTypes.string.isRequired,
   iconName: PropTypes.string,
   iconSrc: PropTypes.string,
+  className: PropTypes.string,
+  title: PropTypes.string,
+  chip: PropTypes.bool,
```
Comment (georgewrmarshall) [inline ui/components/ui/site-origin/site-origin.js:43]: Would be good to add some propType descriptions. Not exactly sure what title does? Was it just added to ensure compatibility [here](https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/app/connected-sites-list/connected-sites-list.component.js#L38)? 

Maybe something like this:

```suggestion
/**
   * The origin of the site generally the URL
   */
  siteOrigin: PropTypes.string.isRequired,
  /**
   * Icon name is used for the fallback icon and will display the first letter of the string
   * when iconSrc errors or is not provided.
   */
  iconName: PropTypes.string,
  /**
   * The image src of the icon
   */
  iconSrc: PropTypes.string,
  /**
   * Additional className added to the root html element
   */
  className: PropTypes.string,
  /**
   * title specifies extra information and is added to a title attribute added to the root html element
   * It is most often shown as a tooltip text when the mouse moves over the element.
   */
  title: PropTypes.string,
  /**
   * Whether the SiteOrigin is displayed as a chip or plan text.
   * if false iconSrc and iconName props will not be used.
   */
  chip: PropTypes.bool,
```

File: ui/components/ui/site-origin/site-origin.js [hunk 5]
```diff
@@ -1,16 +1,35 @@
...
+      {chip ? (
+        <Chip
+          label={siteOrigin}
+          maxContent={false}
+          leftIcon={
+            <IconWithFallback
+              icon={iconSrc}
+              name={iconName}
+              size={32}
+              className={className}
```
Comment (georgewrmarshall) [inline ui/components/ui/site-origin/site-origin.js:26]: Should this be here? Additional className should usually be added to the root element?
```suggestion
```

**Review by georgewrmarshall — COMMENTED** [review]
Nice work! We should probably standardize the display of this component a bit more. Will add to the design system backlog. Just a couple suggestions.

File: ui/components/ui/site-origin/site-origin.js (hunk 5 above)
Comment (adonesky1) [reply ui/components/ui/site-origin/site-origin.js:26]: Yup whoops!

**Review by georgewrmarshall — APPROVED** [review]
LGTM! Nice work 💯

**Comment by adonesky1** [comment]
@darkwing could you take another look when you have a chance?

File: ui/pages/confirm-encryption-public-key/confirm-encryption-public-key.scss [hunk 6]
```diff
@@ -198,6 +198,7 @@
       flex-direction: column;
       align-items: center;
       flex: 1;
+      width: 100%;
```
Comment (digiwand) [inline ui/pages/confirm-encryption-public-key/confirm-encryption-public-key.scss:201]: This is not a blocker. Just wondering, what prompted you to add this in?

**Review by digiwand — APPROVED** [review]
LGTM again! 😄 added 1 non-blocking question

File: ui/pages/confirm-encryption-public-key/confirm-encryption-public-key.scss (hunk 6 above)
Comment (adonesky1) [reply ui/pages/confirm-encryption-public-key/confirm-encryption-public-key.scss:201]: With a sufficiently long origin, the section was extending past 100% (beyond the borders of the notification window)

Comment (digiwand) [reply ui/pages/confirm-encryption-public-key/confirm-encryption-public-key.scss:201]: Thanks for the explanation! 👍

**Pull request merged**